### PR TITLE
Slack notifier

### PR DESCRIPTION
**Pull request by 0x29a**

### CODE REVIEW

File: cloud_control/__init__.py [hunk 1]
```diff
@@ -245,12 +293,37 @@ def perform_command(self):
...
             self._log_and_send_metrics(self.args.command, success)
 
+    def _send_notification(self, message, result=None):
+        """
+        If notification system enabled, collects required information and
+        sends notification.
+        """
+        if self.notification_system:
+            deployment_target = (
+                "GCP Stackdriver" if self.args.command == "deploy" else "Github"
```
[the remark this answers is not in the document]
Comment (0x29a) [reply cloud_control/__init__.py]: @gbmeuk 
Oh, sure, there should be just `GCP`.

File: cloud_control/__init__.py [hunk 2]
```diff
@@ -245,12 +301,37 @@ def perform_command(self):
...
             self._log_and_send_metrics(self.args.command, success)
 
+    def _send_notification(self, message, result=None):
+        """
+        If notification system enabled, collects required information and
+        sends notification.
+        """
+        if self.notification_system:
+            deployment_target = (
+                "Google Cloud Platform" if self.args.command == "deploy" else "Github"
```
Comment (gbmeuk) [inline cloud_control/__init__.py]: This could be some other provider than gcp

File: reporter/base.py [hunk 3]
```diff
@@ -174,3 +173,57 @@ def prepare_metrics(self) -> dict:
...
+        """
+        raise NotImplementedError
+
+    def _send_notification(self, notification_text: str):
+        """
+        Should use notification client to send notification text
+        """
+        raise NotImplementedError
+
+    def send_notification(self, notification: Notification):
```
Comment (gbmeuk) [inline reporter/base.py]: Thing missing here?

File: tests/reporter/test_slack.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    Tests, that slack web client instantiated with correct credentials, and
+    accepts correctly formatted message.
+    """
+    notification_client = mocker.patch("reporter.slack.slack").WebClient
+
+    command_line_args.slack_channel = "general"
+    command_line_args.slack_token = "token"
+
+    notificator = SlackNotificator(command_line_args)
```
Comment (gbmeuk) [inline tests/reporter/test_slack.py]: Notifier. Notificator isn’t a word

**Pull request merged**